### PR TITLE
Metrics fake client fix

### PR DESCRIPTION
**Pull request by niuyueyang1996**
#### What type of PR is this?
/kind bug

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```
NONE

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

fix bug fake metrics client,old client list metrics will return empty,because resources is nodemetricses/podmetricses

### CODE REVIEW

**Review by dgrisonnet — COMMENTED** [review]
Could you please provide a bit more details as to what you are trying to solve here?

File: staging/src/k8s.io/client-go/testing/fixture.go [hunk 1]
```diff
@@ -312,6 +312,11 @@ func (t *tracker) Get(gvr schema.GroupVersionResource, ns, name string) (runtime
 	return obj, nil
 }
 
+var kindToResource = map[string]string{
+	"nodemetricses": "nodes",
+	"podmetricses":  "pods",
```
[the remark this answers is not in the document]
Comment (niuyueyang1996) [reply staging/src/k8s.io/client-go/testing/fixture.go:317]: you can see NodeMetric types.like staging/src/k8s.io/metrics/pkg/apis/metrics/v1alpha1/types.go
Kind is NodeMetric,but resources is node.
```
// +genclient
// +resourceName=nodes
// +genclient:readonly
// +genclient:nonNamespaced
// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object

// NodeMetrics sets resource usage metrics of a node.
type NodeMetrics struct {
	metav1.TypeMeta   `json:",inline"`
	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`

	// The following fields define time interval from which metrics were
	// collected from the interval [Timestamp-Window, Timestamp].
	Timestamp metav1.Time     `json:"timestamp" protobuf:"bytes,2,opt,name=timestamp"`
	Window    metav1.Duration `json:"window" protobuf:"bytes,3,opt,name=window"`

	// The memory usage is the memory working set.
	Usage v1.ResourceList `json:"usage" protobuf:"bytes,4,rep,name=usage,casttype=k8s.io/api/core/v1.ResourceList,castkey=k8s.io/api/core/v1.ResourceName,castvalue=k8s.io/apimachinery/pkg/api/resource.Quantity"`
}
```

if not solve this problem,fake client will always find empty.

File: staging/src/k8s.io/metrics/pkg/client/clientset_test/clientset_test.go [hunk 2]
```diff
@@ -40,3 +45,80 @@ func TestFakeList(t *testing.T) {
 		t.Errorf("Unexpected error: %v", err)
 	}
 }
+func TestFakeMetrics(t *testing.T) {
```
Comment (niuyueyang1996) [inline staging/src/k8s.io/metrics/pkg/client/clientset_test/clientset_test.go:48]: in this test,if not fix the problem
list metrics will return emoty,then will errorf list pod metrics number is not match 5,0

**Comment by dgrisonnet** [comment]
/remove-lifecycle stale

I'll try to have another look at this PR asap.

**Pull request closed without merging**